test/dashboard/index.spec.js

```diff
@@ -5,8 +5,48 @@ require("../base.spec");
 const Dashboard = require("../../dashboard");
 
 describe("dashboard", () => {
+  let dashboard;
+  let dashboardWithOptions;
+  const options = {
+    color: "red",
+    minimal: true,
+    title: "my-title"
+  };
+
+  beforeEach(() => {
+    dashboard = new Dashboard();
```

Comment:
I'd actually remove this `beforeEach` and have just two specs:

```js
it("can create a new no option dashboard", () => {
  const dashboard = new Dashboard();
  // ... all the dashboard asserts.
});

it("can create a new with options dashboard", () => {
  const dashboard = new Dashboard(options);
  // ... all the dashboard with options asserts.
});
```

and combine all the asserts for like objects there...

Reply:
Being new to Cil testing I would love to hear your take on writing the test one way over the other?

Reply:
I tend to look at it as the unit of a test is:

1. Setup
2. State / changes
3. Asserts

Here, we don't have state / changes, we just have (1) setup (`new Dashboard()`) and asserts `expect()`. I think the best approach is combine these. So, for example, could look like for these tests are:

```js
describe("dashboard", () => {
  it("can create a new no option dashboard", () => {
    const dashboard = new Dashboard();

    expect(dashboard).to.be.ok;
    expect(dashboard.color).to.equal("green");
    expect(dashboard.minimal).to.be.false;
    expect(dashboard.stats).to.be.null;
  });

  it("can create a new with options dashboard", () => {
    const dashboardWithOptions = new Dashboard(options);

    expect(dashboardWithOptions.color).to.equal("red");
    expect(dashboardWithOptions.minimal).to.be.true;
  });
});
```

One other thing, is maybe use the `property` assert for better failure messages like:

```js
describe("dashboard", () => {
  it("can create a new no option dashboard", () => {
    const dashboard = new Dashboard();

    expect(dashboard).to.be.ok;
    expect(dashboard).to.have.property("color", "green");
    expect(dashboard).to.have.property("minimal", false);
    expect(dashboard).to.have.property("stats", null);
  });

  it("can create a new with options dashboard", () => {
    const dashboardWithOptions = new Dashboard(options);

    expect(dashboardWithOptions).to.be.ok;
    expect(dashboardWithOptions).to.have.property("color", "red");
    expect(dashboardWithOptions).to.have.property("minimal", true);
  });
});
```

Reply:
Cool beans thank you.